### PR TITLE
todo and forge with tab

### PR DESCRIPTION
**Pull request by hannibal00212**
Adding option to only show todo list and forge overlay when pressing the tab key

### CODE REVIEW

File: src/main/java/io/github/moulberry/notenoughupdates/overlays/MiningOverlay.java [hunk 1]
```diff
@@ -440,11 +441,13 @@ public void update() {
 			}
 			if (forgeDisplay) {
 				overlayStrings = new ArrayList<>();
-				if (NotEnoughUpdates.INSTANCE.config.mining.forgeDisplayEnabledLocations == 1 &&
-					!SBInfo.getInstance().isInDungeon) {
-					overlayStrings.addAll(getForgeStrings(profileConfig.forgeItems));
-				} else if (NotEnoughUpdates.INSTANCE.config.mining.forgeDisplayEnabledLocations == 2) {
-					overlayStrings.addAll(getForgeStrings(profileConfig.forgeItems));
+				if (!NotEnoughUpdates.INSTANCE.config.mining.forgeDisplayOnlyShowTab || Keyboard.isKeyDown(Keyboard.KEY_TAB)) {
```
Comment (NopoTheGamer) [inline src/main/java/io/github/moulberry/notenoughupdates/overlays/MiningOverlay.java:444]: what if someone has changed the keybind

Comment (hannibal00212) [reply src/main/java/io/github/moulberry/notenoughupdates/overlays/MiningOverlay.java:444]: I wrote "tab key" not "opening the tab list" so I dodged a bullet there. Do you know how to access the vanilla minecraft keybild?

Comment (Ascynx) [reply src/main/java/io/github/moulberry/notenoughupdates/overlays/MiningOverlay.java:444]: GameSetting.keyBindPlayerList

Comment (hannibal00212) [reply src/main/java/io/github/moulberry/notenoughupdates/overlays/MiningOverlay.java:444]: This does not react instantly
`Minecraft.getMinecraft().gameSettings.keyBindPlayerList.isKeyDown()`

but this does.
`Keyboard.isKeyDown(Minecraft.getMinecraft().gameSettings.keyBindPlayerList.getKeyCode())`

strange world i guess?

Comment (romangraef) [reply src/main/java/io/github/moulberry/notenoughupdates/overlays/MiningOverlay.java]: KeyBind#isKeyDown vs Keyboard#isKeyDown is also checking for collisions i think

File: src/main/java/io/github/moulberry/notenoughupdates/listener/NEUEventListener.java [hunk 2]
```diff
@@ -149,6 +146,17 @@ public void onTick(TickEvent.ClientTickEvent event) {
 			}
 		}
 
+		//MiningOverlay and TimersOverlay need real tick speed
+		if (neu.hasSkyblockScoreboard()) {
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/listener/NEUEventListener.java:150]: NEUs slow (and centralized) ticking is starting to bother me in places. Perhaps in the future we could work on an overall more reactive coding style. (This seems like a fine bodge tho)

File: src/main/java/io/github/moulberry/notenoughupdates/options/seperateSections/Mining.java [hunk 3]
```diff
@@ -207,6 +207,16 @@ public class Mining {
 	@ConfigAccordionId(id = 2)
 	public int forgeDisplayEnabledLocations = 0;
 
+	@Expose
+	@ConfigOption(
+		name = "Forge Tab",
+		desc = "Only show the forge display when tab list is open\n" +
+			"\u00A7cThis only works outside of Dwarven Caves!"
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/options/seperateSections/Mining.java:214]: Not sure if Dwarven Caves is right

**Review by romangraef — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/options/seperateSections/Mining.java (hunk 3 above)
Comment (jani270) [reply src/main/java/io/github/moulberry/notenoughupdates/options/seperateSections/Mining.java:214]: Dwaven Caves is used by hypixel to descripe both Mines and Crystal Hollows 

See usage of Royal Pigeon outside of Dwarven Caves
image

**Review by romangraef — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Pull request merged**